### DISCUSSION TITLE
"There has been no past successful build for this stage"

### DISCUSSION THREAD

**Discussion by jmichas** [rfennell/AzurePipelines#1189] [Q&A] [answered]
I am using the checkStage, overrideStageName, tag, and checkForManuallyLinkedWI to try and use a previous successful build to lookup WIs from.

I use git tag -l to get the latest tag matching my pattern and have verified it is the last successful build. 

I am not sure why the task does not find a successful build. The overrideStageName is the stage that releases my app to the appstore, my build is only tagged if the pipeline successfully completes. 

How do I fix this?

In the log I get this at the beginning:
````
Only considering builds with the tag(s) 'iOS_SBX_v1.0.119-alpha+805595dc22'
Overriding current stage 'checkout_and_install_tools' with 'release_to_appstore'
Getting items associated the builds since the last successful build to the stage 'release_to_appstore'
````
later in the log I see this:
````
There has been no past successful build for this stage, we need to get the details for all past builds
````

I should add that I get the correct WIs in my notes but I think it takes a really long time because I am bringing back a ton of WIs that end up getting filtered by if statements in my template.

**Comment by rfennell** [comment, marked as answer]
The `tag` parameter for the task is not referring to git tags, but an [Azure DevOps Pipeline Tags](https://elanderson.net/2020/04/azure-devops-pipelines-naming-and-tagging/), I think that could explain what you are seeing

**Reply by jmichas** [reply]
So when you use the tag on successful build in "Get Sources" in the pipeline triggers YAML tab, that is setting a git tag then?

**Reply by rfennell** [reply]
Yes it is, so definitely confusing

**Reply by jmichas** [reply]
Is the default tag for a pipeline the buildnumber? Or do I have to use a task to tag the pipeline?

**Reply by rfennell** [reply]
You add them via the UI using the ellipsis menu on the right

![image](https://user-images.githubusercontent.com/324718/148981455-80b06a30-a163-445b-b26b-c0223c4bdaea.png)

They then appear like this

![image](https://user-images.githubusercontent.com/324718/148981562-899884cf-6699-4da7-b558-a763ad849c58.png)

These tags are for the Azure Pipelines only (they are not the same ones as used for source control or work item)

To add a build tag from with  a pipelines use [this extension from Colin Dembovsky](https://marketplace.visualstudio.com/items?itemName=colinsalmcorner.colinsalmcorner-buildtasks)

**Reply by jmichas** [reply]
Ok great thanks for your help!
